### PR TITLE
docs(design-foundations): Adds motion doc.

### LINKED ISSUE #3957: Add initial motion docs.
Add a new page to pf.org to contain motion principles and guidelines. This should go live with the first set of animations.

### PR DESCRIPTION
**Pull request by edonehoo**
Closes #3957

### CODE REVIEW

**Review by nicolethoen — APPROVED** [review]

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    </CardBody>
+    </Card>
+</GridItem>
+</Grid>
+
+## Motion in PatternFly
+
+PatternFly components support motion in the form of **micro animations**, which are small, unobtrusive, and often understated. These animations serve specific purposes within a UI, and are designed to be subtle and complementary. 
+
+This behavior is built into PatternFly, so there is nothing additional you need to do to support motion. To meet accessibility requirements, reduced motion support is determined based on users' browser settings.
```
Comment (srambach) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md:77]: This isn't universally true. Some things are built in - like the background hover transition on a button. Other things may need to be added ad hoc, like applying a rotation to a cog icon. We'd supply the class that implements the rotation, but the consumer might need to specifically add it. It's a little TBD still :-) 

For the second sentence, I think I'd say something along the lines of "Wherever PatternFly provides animation, reduced motion support (based on user's browser settings) is provided."
(which isn't quite true yet with those that existed in pf5)

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+#### Delay 
+Delay tokens specify the length of time that should pass before an animation begins. Delay options include none, short, default, and long.
+
+#### Timing 
+Timing-function tokens specify the easing path that an animation takes. These paths are defined by cubic bezier curves, which define the start and end points ... [fact check]
+
+Options include: 
+- **Accelerate:** Use to specify when an animation should begin to speed up.
+- **Default:** Use to specify when an animation should maintain a cosistent speed.
```
Comment (srambach) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md:140]: "consistent" 😄

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+A **motion pattern** is a combination of tokens and properties that work together to solve common animation needs. These patterns help ensure that the behavior of a type of animation is consistent across similar use cases—even if different components are involved.
+
+For example, an "action fade" pattern may dictate how animation is applied when a user selects an action. This pattern specifies the timing and duration of the fade, so that the animation is the same across different action components. In this example, we would want users to see the same hover/click animation whether they're interacting with an action button, an action menu, and so on.
+
+### Custom motion 
+
+For some scenarios, like creating a new [PatternFly extension](/extensions/about-extensions), you may need to customize motion that doesn't already exist in PatternFly components. When you're creating a new animation behavior, make sure that:
+- There is no existing support for the animation.
+- The animation adheres to our motion principles. 
+- Your implementation uses appropriate semantic tokens.
```
Comment (srambach) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md:154]: Add that the user's setting for `prefers-reduced-motion` should be respected? Unless that is expected as part of adhering to our motion principles.

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+---
+
+import { Alert, AlertActionLink, Card, CardHeader, CardTitle, CardBody, CardFooter, Grid, GridItem} from '@patternfly/react-core';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/esm/icons/external-link-alt-icon';
+
+<Alert 
+variant="info" 
+title="Under development" 
+actionLinks={
+    <AlertActionLink component="a" href="#">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md:13]:
```suggestion
    <AlertActionLink component="a" href="https://github.com/orgs/patternfly/projects/16/views/18">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+| Slide-in |  |
+| Fade |  |
+| Icon |  |
+<br />
+
+#### Delay 
+Delay tokens specify the length of time that should pass before an animation begins. Delay options include none, short, default, and long.
+
+#### Timing 
+Timing-function tokens specify the easing path that an animation takes. These paths are defined by cubic bezier curves, which define the start and end points ... [fact check]
```
Comment (srambach) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md:136]: Do you want to confirm this before merging this update?

Comment (edonehoo) [reply packages/documentation-site/patternfly-docs/content/design-guidelines/styles/motion/motion.md]: yes, ty for pointing out!

**Review by srambach — APPROVED** [review]
🎢 🚀

**Review by dlabaj — APPROVED** [review]
LGTM

**Pull request merged**